File: src/contests/cg/puzzle/horse-racing-duals.ts
// 🎮 CodinGame Puzzle - horse-racing-duals
// https://www.codingame.com/training/easy/horse-racing-duals

const n = parseInt(readline())
const strengths: number[] = []
for (let i = 0; i < n; i++) {
  strengths.push(parseInt(readline()))
}

strengths.sort((a, b) => a - b)

let minDiff = Infinity
for (let i = 1; i < strengths.length; i++) {
  const diff = strengths[i] - strengths[i - 1]
  if (diff < minDiff) {
    minDiff = diff
  }
}

console.log(minDiff + 0)

File: src/contests/cg/puzzle/ruby/horse-racing-duals.rb
# 🎮 CodinGame Puzzle - horse-racing-duals
# https://www.codingame.com/training/easy/horse-racing-duals

n = gets.to_i
strengths = Array.new(n) { gets.to_i }.sort
min_diff = Float::INFINITY
(1...strengths.length).each do |i|
  diff = strengths[i] - strengths[i - 1]
  min_diff = diff if diff < min_diff
end
puts min_diff
